### PR TITLE
Fizz buzz tree

### PR DESCRIPTION
**Pull request by tylersayvetz**
- [x] write tests
- [x] update all readmes
- [ ] uml diagram . :(

### CODE REVIEW

File: code-challenges/tree/fizzTree.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    if (current.left) this.inOrder(current.left, output)
+    output.unshift(current.value)
+    if (current.right) this.inOrder(current.right, output)
+    return output
+  }
+
+}
+
+function fizzBuzzTree(tree) {
+  const newRootValue = fizzify(tree.root.value)
```
Comment (XiaojuanCai) [inline code-challenges/tree/fizzTree.js]: What happens if tree.root is null?

File: code-challenges/tree/fizzTree.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const newTree = new BinaryTree(new Node(newRootValue))
+  
+  duplicateTree(tree.root, newTree.root)
+
+  //traverse both trees at once, attaching more nodes to our new tree as we go.
+  function duplicateTree(currA, currB) {
+    if (!currA) return 
+    if (currA.left) {
+      currB.left = new Node(fizzify(currA.left.value));
```
Comment (XiaojuanCai) [inline code-challenges/tree/fizzTree.js]: currB.left = new Node(fizzify(currA.left.value));
currB.right = new Node(fizzify(currA.right.value));
emmm, seems they are doing something very similar to each other. Is there a better way write this function to make it DRY?

File: code-challenges/tree/fizzTree.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+function fizzBuzzTree(tree) {
+  const newRootValue = fizzify(tree.root.value)
+  const newTree = new BinaryTree(new Node(newRootValue))
+  
+  duplicateTree(tree.root, newTree.root)
+
+  //traverse both trees at once, attaching more nodes to our new tree as we go.
+  function duplicateTree(currA, currB) {
+    if (!currA) return
```
Comment (XiaojuanCai) [inline code-challenges/tree/fizzTree.js]: Do you think your code will ever reach a state that currA is null?

File: code-challenges/tree/README.md [hunk 4]
```diff
@@ -20,7 +20,12 @@ For time, all methods are at worst case O(n) time. None are N**2. At best case (
 For space, the traversal methods all have O(n) complexity. This is because I build an output array that is exaclty as long as N nodes. 
 
 
+## FizzBuzz files
 
+### API 
+`fizzify ` takes a numerical value and returns 'fizz' 'buzz' or 'Fizzbuzz', depending on if it dvisible by 3, 5, and both3 && 5, respectively. 
+
+`fizzBuzzTree` takes in a binary tree root and returns a copy of the tree, with fizz and buzz replaced in the proper spots, according to fizzify.
```
Comment (adrianhuebner) [inline code-challenges/tree/README.md]: Can you please add a whitebaord image to your readme

**Pull request closed without merging**